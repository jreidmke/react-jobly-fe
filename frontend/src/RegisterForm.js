import {useState} from 'react'
import {useHistory, Link} from 'react-router-dom';
import JoblyApi from './api';

const RegisterForm = ({register}) => {
    const history = useHistory();
    const [formData, setFormData] = useState({
        username: "",
        password: "",
        firstName: "",
        lastName: "",
        email: ""
    });

    const handleChange = e => {
        const { name, value } = e.target;
        setFormData(formData => ({
            ...formData,
            [name]: value
        }))
    };

    const submit = async(e) => {
        e.preventDefault();
        let res = await register(formData);
        if(res.success) {
            history.push('/companies');
        } else {
            history.push('/');
        }
        // const token = await JoblyApi.register(formData);
        // localStorage.setItem('token', token);
    }

    return(
        <div>
            <form onSubmit={submit}>

                <input
                onChange={handleChange}
                type="text"
                name="username"
                id="username"
                value={formData.username}
                placeholder="Username"
                />
                
                <input
                onChange={handleChange}
                type="password"
                name="password"
                id="password"
                value={formData.password}
                placeholder="Password"
                />

                <input
                onChange={handleChange}
                type="text"
                name="firstName"
                id="firstName"
                value={formData.firstName}
                placeholder="First Name"
                />

                <input
                onChange={handleChange}
                type="text"
                name="lastName"
                id="lastName"
                value={formData.lastName}
                placeholder="Last Name"
                />

                <input
                onChange={handleChange}
                type="text"
                name="email"
                id="email"
                value={formData.email}
                placeholder="Email Address"
                />

                <button>Submit</button>

            </form>
        </div>
    )
}

export default RegisterForm;