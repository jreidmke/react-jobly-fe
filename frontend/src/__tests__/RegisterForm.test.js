import React from "react";
import { render } from "@testing-library/react";
import RegisterForm from "../RegisterForm";
import { MemoryRouter } from "react-router";

it("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <RegisterForm />
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});
