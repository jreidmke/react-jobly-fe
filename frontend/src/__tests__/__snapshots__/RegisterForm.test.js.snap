// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <input
        id="username"
        name="username"
        placeholder="Username"
        type="text"
        value=""
      />
      <input
        id="password"
        name="password"
        placeholder="Password"
        type="password"
        value=""
      />
      <input
        id="firstName"
        name="firstName"
        placeholder="First Name"
        type="text"
        value=""
      />
      <input
        id="lastName"
        name="lastName"
        placeholder="Last Name"
        type="text"
        value=""
      />
      <input
        id="email"
        name="email"
        placeholder="Email Address"
        type="text"
        value=""
      />
      <button>
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
